src/main/java/net/openhft/chronicle/network/tcp/ChronicleSocketChannel.java

```diff
@@ -39,4 +40,18 @@
     void setOption(SocketOption<Boolean> soReuseaddr, boolean b) throws IOException;
 
     SocketChannel socketChannel();
+
+    /**
+     * Is the passed exception one that results from reading from or writing to a reset
+     * connection?
+     *
+     * @param e The exception
+     * @return true if the exception is one that signifies the connection was reset
+     */
+    static boolean isAConnectionResetException(IOException e) {
+        String message = e.getMessage();
+        return message != null &&
+                (message.startsWith("Connection reset by peer")
+                        || e instanceof SocketException && message.equals("Connection reset"));
+    }
```

Comment:
Not sure if this is the best home for it, but I didn't want to create a "NetworkUtils" class. Happy for any bike-shedding about where this should live.

Reply:
I would introduce a utility class in the `internal` package in order to keep the API surface under control.

Reply:
Good call, done

Reply:
In the examples visible in this PR, there is no fundamentally different flow control when this function returns true.
ie, we either continue to log, or to drop a connection etc. The only difference switching on this function is logging.
In which case, why don't we just log the message regardless?

We know any scraping of the message content is fragile, and something like this is only likely to get (ab)used over time.

If there's a counter example where actual flow changes (eg a reconnect attempt vs a rethrow) then can you please show. Else I would suggest this is removed in favour of just logging the message.

Reply:
The difference is the level things are logged at, when the IOException is due to the connection being reset we log at TRACE level, but when it's anything else (IOException covers a lot as you know) we treat it more seriously and log at WARN level. I discovered this behaviour because a bunch of tests became flaky due to these warnings.

It's the change in TCPEventHandler that drove this, I just incorporated it in the other place (where I agree you could just log the message) because the logic was repeated.

Reply:
Perhaps a change so that both TCPChannelHub and TCPEventHandler log at debug level when it's connection reset? I don't know why we'd want to warn about that, it happens fairly often.
If we log at warn level for connection reset we need to add ignoreException in pretty much every test that involves networking.